packages/cf.js/src/utils/serializer.ts

```diff
@@ -28,7 +26,7 @@ const deserializeArray = data => data.map(value => deserialize(value));
 
 const deserializeBigNumber = data => ethers.utils.bigNumberify(data._hex);
 
-const deserializeSignature = data => new Signature(data.v, data.r, data.s);
+const deserializeSignature = data => data;
```

Comment:
Wouldn't we want to instantiate an ethers Signature using the data being passed in? Otherwise this deserialization seems to just pass through the data that _would be_ in the signature but not an ethers Signature instance itself.

Reply:
ethers.utils.Signature is just a Typescript interface, not a class. Anything with a v, r and s property is considered an ethers.utils.Signature

Reply:
Got it - in which case @joelalejandro's comment is more useful here